Detectors/GlobalTracking/src/MatchGlobalFwdAssessment.cxx

```diff
@@ -626,48 +624,48 @@ void GloFwdAssessment::finalizePurityAndEff()
+    hPairingEffInner->Divide(PairablePtProjInner); // Pairing Efficiency = N_reco / N_Pairable
+    hPairingEffInner->SetNameTitle(Form("GMTrackPairingEffInnerPtCut_%.2f", scoreCut), Form("%.2f cut", scoreCut));
+    hPairingEffInner->GetYaxis()->SetTitle("Pairing Efficiency [ N_{Rec} / N_{pairable}]");
+    hPairingEffInner->SetOption("COLZ");
+    hPairingEffInner->SetMarkerStyle(kFullCircle);
+    hPairingEffInner->SetMinimum(0.0);
+    hPairingEffInner->SetMaximum(1.8);
 
     // Outer pseudorapidity
     auto RecoPtProjOuter = (TH1*)Reco->ProjectionX(Form("_OuterRecoCut_%.2f", scoreCut), minBin, midBin, 0, maxScoreBin);
     auto TruePtProjOuter = (TH1*)hTrue->ProjectionX(Form("_OuterTrueCut_%.2f", scoreCut), minBin, midBin, 0, maxScoreBin);
 
-    auto& hPOuter = mPurityPtOuterVecTH2.emplace_back((std::unique_ptr<TH2D>)static_cast<TH2D*>(TruePtProjOuter->Clone()));
-    hPOuter->Divide(RecoPtProjOuter);
-    hPOuter->SetNameTitle(Form("TH2GMTrackPurityOuterEtaCut_%.2f", scoreCut), Form("%.2f cut", scoreCut));
-    hPOuter->GetYaxis()->SetTitle("Pairing Purity [ N_{True} / N_{Rec}]");
-    hPOuter->SetOption("COLZ");
-    hPOuter->SetMarkerStyle(kFullTriangleUp);
-    hPOuter->SetMinimum(0.0);
-    hPOuter->SetMaximum(1.2);
-
-    auto& hOuter = mPairingPtOuterVecTH1.emplace_back((std::unique_ptr<TH1D>)static_cast<TH1D*>(RecoPtProjOuter->Clone()));
-    hOuter->Divide(PairablePtProjInner);
-    hOuter->SetNameTitle(Form("GMTrackPairingEffOuterPtCut_%.2f", scoreCut), Form("%.2f cut", scoreCut));
-    hOuter->GetYaxis()->SetTitle("Pairing Efficiency [ N_{Rec} / N_{pairable}]");
-    hOuter->SetOption("COLZ");
-    hOuter->SetMarkerStyle(kFullTriangleUp);
-    hOuter->SetMinimum(0.0);
-    hOuter->SetMaximum(1.8);
+    auto& hPurityOuter = mPurityPtOuterVecTH2.emplace_back((std::unique_ptr<TH2D>)static_cast<TH2D*>(TruePtProjOuter->Clone()));
+    hPurityOuter->Divide(RecoPtProjOuter); // Global Pairing Purity = N_true / N_reco
+    hPurityOuter->SetNameTitle(Form("TH2GMTrackPurityOuterEtaCut_%.2f", scoreCut), Form("%.2f cut", scoreCut));
+    hPurityOuter->GetYaxis()->SetTitle("Pairing Purity [ N_{True} / N_{Rec}]");
+    hPurityOuter->SetOption("COLZ");
+    hPurityOuter->SetMarkerStyle(kFullTriangleUp);
+    hPurityOuter->SetMinimum(0.0);
+    hPurityOuter->SetMaximum(1.2);
+
+    auto& hPairingEffOuter = mPairingPtOuterVecTH1.emplace_back((std::unique_ptr<TH1D>)static_cast<TH1D*>(RecoPtProjOuter->Clone()));
+    hPairingEffOuter->Divide(PairablePtProjInner); // Pairing Efficiency = N_reco / N_Pairable
+    hPairingEffOuter->SetNameTitle(Form("GMTrackPairingEffOuterPtCut_%.2f", scoreCut), Form("%.2f cut", scoreCut));
+    hPairingEffOuter->GetYaxis()->SetTitle("Pairing Efficiency [ N_{Rec} / N_{pairable}]");
+    hPairingEffOuter->SetOption("COLZ");
+    hPairingEffOuter->SetMarkerStyle(kFullTriangleUp);
+    hPairingEffOuter->SetMinimum(0.0);
+    hPairingEffOuter->SetMaximum(1.8);
```

Comment:
For the record: the projections like RecoPtProjOuter are all leaking. You should delete this which are not needed (or make them unique_ptrs). BTW, instead of calling `Clone` which is extremely expensive call, you could do directly
```
auto& hPairingEffOuter = mPairingPtOuterVecTH1.emplace_back();
hPairingEffOuter.reset(RecoPtProjOuter);
```

Reply:
Hi @shahor02 , thanks for the input. This part of the code is not executed in production, so it will not cause problems. Given that @RenEjima is ready to open a PR that depends on this one, may I ask to have this merged and fix memory leak and optimization in a separate PR?